controlplane/eks/controllers/awsmanagedcontrolplane_controller_unit_test.go

```diff
@@ -61,3 +61,47 @@ func TestSecurityGroupRolesForCluster(t *testing.T) {
 		})
 	}
 }
+
+func TestGetUniqueRolesFromLists(t *testing.T) {
+	tests := []struct {
+		name                        string
+		rolesFromMachinePools       []string
+		rolesFromMachineDeployments []string
+		expected                    []string
+	}{
+		{
+			"check duplicates",
+			[]string{
+				"nodes.cluster-api-provider-aws.sigs.k8s.io",
+				"eks-nodes.cluster-api-provider-aws.sigs.k8s.io",
+			},
+			[]string{
+				"nodes.cluster-api-provider-aws.sigs.k8s.io",
+			},
+			[]string{
+				"nodes.cluster-api-provider-aws.sigs.k8s.io",
+				"eks-nodes.cluster-api-provider-aws.sigs.k8s.io",
+			},
+		},
+		{
+			"unique roles",
+			[]string{
+				"eks-nodes.cluster-api-provider-aws.sigs.k8s.io",
+			},
+			[]string{
+				"other-nodes.cluster-api-provider-aws.sigs.k8s.io",
+			},
+			[]string{
+				"eks-nodes.cluster-api-provider-aws.sigs.k8s.io",
+				"other-nodes.cluster-api-provider-aws.sigs.k8s.io",
+			},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			g := NewWithT(t)
+			fromRun := getUniqueRolesFromList(tt.rolesFromMachinePools, tt.rolesFromMachineDeployments)
+			g.Expect(fromRun).To(Equal(tt.expected), "value from getUniqueRolesFromList does not match expected")
+		})
+	}
+}
```

Comment:
```suggestion
```

Reply:
Now that this test is obsolete though, you must write one to test the new code part in another way. :) Sorry about that. :)

Reply:
No worries, I am thinking of writing a test that leverages envtest.

Reply:
added the tests

Reply:
Nice :)